src/transitions/Grow.js

```diff
@@ -0,0 +1,172 @@
+// @flow
+
+import React, { PureComponent } from 'react';
+import type { Element } from 'react';
+import withTheme from '../styles/withTheme';
+import Transition from '../internal/Transition';
+import type { TransitionCallback } from '../internal/Transition';
+
+export function getScale(value: number) {
+  return `scale(${value}, ${value ** 2})`;
+}
+
+type DefaultProps = {
+  theme: Object,
+  transitionDuration: 'auto',
+};
+
+export type Props = {
+  /**
+   * The content of the component.
+   */
+  children?: Element<*>,
+  /**
+   * Callback fired before the component is entering
+   */
+  onEnter?: TransitionCallback,
+  /**
+   * Callback fired when the component is entering
+   */
+  onEntering?: TransitionCallback,
+  /**
+   * Callback fired when the component has entered
+   */
+  onEntered?: TransitionCallback, // eslint-disable-line react/sort-prop-types
+  /**
+   * Callback fired before the component is exiting
+   */
+  onExit?: TransitionCallback,
+  /**
+   * Callback fired when the component is exiting
+   */
+  onExiting?: TransitionCallback,
+  /**
+   * Callback fired when the component has exited
+   */
+  onExited?: TransitionCallback, // eslint-disable-line react/sort-prop-types
+  /**
+   * Use that property to pass a ref callback to the root component.
+   */
+  rootRef?: Function,
+  /**
+   * @ignore
+   */
+  theme?: Object,
+  /**
+   * Set to 'auto' to automatically calculate transition time based on height
+   */
+  transitionDuration?: number | 'auto',
+};
+
+type AllProps = DefaultProps & Props;
+
+/**
+ * @ignore - internal component.
+ */
+class Grow extends PureComponent<DefaultProps, AllProps, void> {
```

Comment:
The component takes a children, PureComponent is harming the performance, let's use to Component.